add a docs page on build filters (#3233)
dart-lang/build: docs/partial_builds.md

# Build Filters

Build filters allow you to choose explicitly which files to build instead of
building entire directories or projects.

A build filter is a combination of a package and a path, with glob syntax
supported for each.

Whenever a build filter is provided, only required outputs matching one of the

> build filters will be built, in addition to the inputs to those outputs.

Reviewer: ```suggestion
build filters will be built, in addition to any required files for those outputs.
```

## Command Line Usage

Build filters are supplied using the `--build-filter` option, which accepts

> relative paths under the package as well as `package:` uris.

Reviewer: [nit] 
```suggestion
relative paths under the root package as well as `package:` uris.
```
